scripts/build.js

```diff
@@ -17,7 +17,12 @@ async function main(argv = process.argv) {
   const projectRoot = path.join(__dirname, '..');
   const buildPath = path.join(projectRoot, 'build');
   const distPath = path.join(projectRoot, 'dist');
-  const gitPath = process.env.GIT_DIR ?? path.join(projectRoot, '.git');
+
+  var gitPath = null;
+  if (process.env.COMMIT_HASH == null) {
+    gitPath = path.join(projectRoot, '.git');
+  }
+
```

Comment:
1. Try to keep this close to where it's used.
2. We generally void using `var`, Always use `const` if you don't intend for the value to change since it will protect you from mistakes generally. use let only if you need to re-assign. `var` works like `let` but hoists the variable to a level above the current scope. 
3. I think the logic flow is a little weird now. Try to simplify it. The main problem is we're being a little implicit. Where the logic is `if a set b; if b set c; if c....` If it's spread out like that it becomes hard to follow.

Reply:
I'd say still have the ability to specify the gitPath like `const gitPath = process.env.GIT_DIR ?? path.join(projectRoot, '.git');` but otherwise seems fine.

Also, reply to review comments if you've addressed them and just resolve them if You feel they've been suitably addressed.

Reply:
That seems like a good idea, given `GIT_DIR` could be used as a manual override. We can just add it as a parameter to `default.nix` and `release.nix`, and set it to some default empty value. That would allow for the user to override the `.git` directory, although I'm unsure whether it will resolve to an appropriate path in the Nix store.

Reply:
I think its good to go now, its got a manual override for GIT_DIR that can be set through changing the `nix-build` parameter.